### PR TITLE
Feature/sharepoint list extension kpi5

### PR DESCRIPTION
**Pull request by marcinpurtak**
## Summary

Source, task and flow for sharepoint list connector updated to handle new types of fields and special characters
Small code improvements
Docstring and parameters update

## Importance

Import for extraction of new fields

## Checklist

This PR:
- [x] follows the guidelines laid out in `CONTRIBUTING.md`
- [ ] links relevant issue(s)
- [x] adds/updates tests (if appropriate)
- [x] adds/updates docstrings (if appropriate)
- [x] adds an entry in `CHANGELOG.md`

### CODE REVIEW

File: viadot/tasks/sharepoint.py [hunk 1]
```diff
@@ -364,7 +429,7 @@ def change_column_name(
 
         # Rename columns names inside DataFrame
         df = df.rename(columns=dictionary)
-
+        # Check again for duplicates
```
[the remark this answers is not in the document]
Comment (marcinpurtak) [reply viadot/tasks/sharepoint.py]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,16 +5,32 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
-### Added
-- Added new view type `agent_interaction_view_type` in `Genesys`source.
+- Modified `SharepointList` source class: 
+  -> docstrings update
+- Modified `SharepointToADLS` flow class:
+  -> docstrings update
+  -> changed key_value_param: bool = False to prevent forced KV store append
+- Modified `SharepointListToADLS` flow class:
+  -> changed key_value_param: bool = False to prevent forced KV store append
+- Modified `SharepointList` source class:
```
Comment (Rafalz13) [inline CHANGELOG.md:24]: `Modified `SharepointList` source class:` is duplicated. You can use it once and fill with the changes

File: viadot/flows/sharepoint_to_adls.py [hunk 3]
```diff
@@ -42,6 +42,7 @@ def __init__(
         if_exists: str = "replace",
         validate_df_dict: dict = None,
         timeout: int = 3600,
+        key_value_param: bool = False,
```
Comment (Rafalz13) [inline viadot/flows/sharepoint_to_adls.py:45]: It is already a similar parameter in BigQuery added: `set_prefect_kv`. What do you think about using the same name here?

File: viadot/flows/sharepoint_to_adls.py [hunk 4]
```diff
@@ -188,37 +192,35 @@ class SharepointListToADLS(Flow):
...
         overwrite_adls: bool = True,
         output_file_extension: str = ".parquet",
         validate_df_dict: dict = None,
+        key_value_param: bool = False,
         *args: List[any],
         **kwargs: Dict[str, Any],
     ):
-        """
-        Run Flow SharepointListToADLS.
+        """_summary_
```
Comment (Rafalz13) [inline viadot/flows/sharepoint_to_adls.py:214]: Summary is missing.

File: viadot/flows/sharepoint_to_adls.py [hunk 5]
```diff
@@ -188,37 +192,35 @@ class SharepointListToADLS(Flow):
...
-                                    All propertys can be found under list.item.properties.
-                                    Default to ["Title"]
-        filters (dict): Dictionary with operators which filters the SharepointList output.
+            name (str): Prefect flow name.
+            list_title (str): Title of Sharepoint List.
+            site_url (str): URL to set of Sharepoint Lists.
+            path (str): Local file path. Default to None.
+            adls_dir_path (str): Azure Data Lake destination folder/catalog path. Defaults to None.
+            adls_file_name (str): Name of file in ADLS. Defaults to None.
+            filters (dict, optional): Dictionary with operators which filters the SharepointList output.
```
Comment (Rafalz13) [inline viadot/flows/sharepoint_to_adls.py:223]: Add `Defaults to False` at the end.

File: viadot/sources/sharepoint.py [hunk 6]
```diff
@@ -101,19 +104,37 @@ def __init__(
         *args,
         **kwargs,
     ):
+        """_summary_
```
Comment (Rafalz13) [inline viadot/sources/sharepoint.py:107]: Summary is missing.

File: viadot/sources/sharepoint.py [hunk 7]
```diff
@@ -132,43 +153,80 @@ def get_connection(
...
-    ):
+    ) -> List:
+        """
+        Function for geting list of fields objects from the sharepoint list.
+        It can get all fields available if required_fields not passed
+        or just the one which are in the list required_fields.
 
-        ctx = self.get_connection(site_url=site_url)
+        Args:
+            list_title (str): name of the sharepoint list
```
Comment (Rafalz13) [inline viadot/sources/sharepoint.py:221]:
```suggestion
            list_title (str): Name of the sharepoint list.
```

File: viadot/sources/sharepoint.py [hunk 8]
```diff
@@ -132,43 +153,80 @@ def get_connection(
...
+            site_url (str): url to the sharepoint list with "/" at the end
```
Comment (Rafalz13) [inline viadot/sources/sharepoint.py:222]:
```suggestion
            site_url (str): URL to the sharepoint list with "/" at the end.
```

File: viadot/sources/sharepoint.py [hunk 9]
```diff
@@ -132,43 +153,80 @@ def get_connection(
...
+            required_fields (List[str], optional ): List of required fields to ingest. It will get all fields if not passed.
 
+        Returns:
+            List: list with office365 sharepoint list field objects
```
Comment (Rafalz13) [inline viadot/sources/sharepoint.py:226]:
```suggestion
            List: List with office365 sharepoint list field objects.
```

File: viadot/sources/sharepoint.py [hunk 10]
```diff
@@ -182,25 +240,42 @@ def get_fields(
...
+        selection of fields with details about possible expansion for more data or details.
+        Field types to extract more values can be: "User*", "MultiChoice"
         field_property to expand can be: ID, Title, FieldTypeKind, TypeAsString and many more.
             -> more properties can be discovered by getting list.item.properties.
-            Default to "Title"
+
+        Args:
+            list_title (str): _description_. Defaults to None.
+            site_url (str): _description_. Defaults to None.
+            required_fields (List[str], optional): _description_. Defaults to None.
```
Comment (Rafalz13) [inline viadot/sources/sharepoint.py:268]: description is missing here.

File: viadot/tasks/sharepoint.py [hunk 11]
```diff
@@ -245,7 +245,7 @@ class SharepointListToDF(Task):
         field_property (List[str]): Property to expand with expand query method.
                                     All propertys can be found under list.item.properties.
                                     Default to ["Title"]
-        filters (dict): Dictionary with operators which filters the SharepointList output.
+        filters (dict, optional): Dictionary with operators which filters the SharepointList output.
```
Comment (Rafalz13) [inline viadot/tasks/sharepoint.py:248]:
```suggestion
        filters (dict, optional): Dictionary with operators which filters the SharepointList output. Defaults to None.
```

File: viadot/tasks/sharepoint.py [hunk 12]
```diff
@@ -313,16 +317,65 @@ def __init__(
...
+
     def _convert_camel_case_to_words(self, input_str: str) -> str:
+        """
+        Function for converting internal names joined as camelCase column names  to regular words
+
+        Args:
+            input_str (str): Column name
+
+        Returns:
+            str: Converted column name
```
Comment (Rafalz13) [inline viadot/tasks/sharepoint.py:377]:
```suggestion
        Function for converting internal names joined as camelCase column names to regular words.

        Args:
            input_str (str): Column name.

        Returns:
            str: Converted column name.
```

File: viadot/tasks/sharepoint.py [hunk 13]
```diff
@@ -331,11 +384,23 @@ def _convert_camel_case_to_words(self, input_str: str) -> str:
...
+        !Warning!
+            Names are taken from field properties Title!
+            Because of that the resulting column name might have different then initial name.
+
+        Args:
+            df (pd.DataFrame): A data frame with loaded column names from sharepoint list.
+            credentials (str): Credentials str for sharepoint connection establishing. Defaults to None.
+
+        Returns:
+            pd.DataFrame: Data frame with changed column names
```
Comment (Rafalz13) [inline viadot/tasks/sharepoint.py:399]:
```suggestion
            pd.DataFrame: Data frame with changed column names.
```

File: viadot/flows/sharepoint_to_adls.py [hunk 14]
```diff
@@ -242,16 +244,27 @@ def __init__(
...
+                                    lead to errors - extension of sp connector would be required.
+                                    Default to ["Title"]. Defaults to "Title".
+            row_count (int, optional): Number of downloaded rows in single request.Defaults to 5000.
+            adls_sp_credentials_secret (str, optional): Credentials to connect to Azure ADLS
+                                    If not passed it will take cred's from your .config/credentials.json Defaults to None.
+            sp_cert_credentials_secret (str, optional): Credentials to verify Sharepoint connection.
+                                    If not passed it will take cred's from your .config/credentials.json Default to None.
+            vault_name (str, optional): KeyVaultSecret name. Default to None.
+            overwrite_adls (bool, optional): Whether to overwrite files in the lake. Defaults to True.
+            output_file_extension (str, optional): _description_. Defaults to ".parquet".
```
Comment (Rafalz13) [inline viadot/flows/sharepoint_to_adls.py:265]: Description is missing here.

File: viadot/sources/sharepoint.py [hunk 15]
```diff
@@ -101,19 +104,37 @@ def __init__(
...
 
-    def get_connection(
-        self,
-        site_url: str = None,
-    ):
+    def get_connection(self, site_url: str):
+        """Function for connecting into Sharepoint with AuthenticationContext
+
+        Args:
+            site_url (str): url of the sharepoint list
```
Comment (Rafalz13) [inline viadot/sources/sharepoint.py:132]:
```suggestion
        """Function for connecting into Sharepoint with AuthenticationContext.

        Args:
            site_url (str): URL of the sharepoint list.
```

File: viadot/sources/sharepoint.py [hunk 16]
```diff
@@ -101,19 +104,37 @@ def __init__(
...
 
-        # Connecting into Sharepoint with AuthenticationContext
+        Returns:
+            ctx: authentication context
```
Comment (Rafalz13) [inline viadot/sources/sharepoint.py:135]: Start with capital letter and at dot at the end.

**Review by Rafalz13 — CHANGES_REQUESTED** [review]
Added few comments. Please add missing docstrings.

File: viadot/flows/sharepoint_to_adls.py (hunk 3 above)
Comment (marcinpurtak) [reply viadot/flows/sharepoint_to_adls.py:45]: Done

File: CHANGELOG.md [hunk 17]
```diff
@@ -5,11 +5,33 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+- Added new view type `agent_interaction_view_type` in `Genesys`source.
+### Changed
+- Splitted test for Eurostat on source tests and task tests
+- Modified `SharepointList` source class: 
+  -> docstrings update
+- Modified `SharepointToADLS` flow class:
+  -> docstrings update
+  -> changed key_value_param: bool = False to prevent forced KV store append
+- Modified `SharepointListToADLS` flow class:
+  -> changed key_value_param: bool = False to prevent forced KV store append
```
Comment (Rafalz13) [inline CHANGELOG.md:24]: Change `key_value_param` to `set_prefect_kv`.

**Review by Rafalz13 — COMMENTED** [review]
Looks good. Only 1 change.

**Review by Rafalz13 — APPROVED** [review]

**Pull request merged**